### PR TITLE
deprecate logm in favor of log

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request deprecates the spelling `logm` for matrix logarithm in favor of `log`. Followup to the much-debated #23233 and #23504. Also ref. #19598 and #8450. Best!

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -338,6 +338,8 @@ Deprecated or removed
 
   * `expm` has been deprecated in favor of `exp` ([#23233]).
 
+  * `logm` has been deprecated in favor of `log` ([#CATS]).
```
Comment (ararslan) [inline NEWS.md]: 🐈 🐈

File: test/linalg/dense.jl [hunk 2]
```diff
@@ -677,7 +677,7 @@ end
     @test exp(a) == exp(a)
     @test isposdef(one(elty))
     @test sqrtm(a) == sqrt(a)
-    @test logm(a) ≈ log(a)
+    @test log(a) ≈ log(a)
```
Comment (Keno) [inline test/linalg/dense.jl]: These tests don't seem particularly useful anymore.

Comment (Sacha0) [reply test/linalg/dense.jl]: Agreed 😎

**Comment by Sacha0** [comment]
Thanks all! :)

**Pull request merged**